### PR TITLE
[RFC] metrics: ci: Add more tests to the metrics ci script

### PR DESCRIPTION
**Pull request by grahamwhaley**
We don't run all the metrics tests for the metrics CI, but we do
want to run a 'good subset' that covers all key data from all key
areas of interest.
Thus, add memory footprint, network and storage tests to the default
set of metrics we run in the ci.

Fixes: #414

Signed-off-by: Graham Whaley <graham.whaley@intel.com>

### CODE REVIEW

**Comment by grahamwhaley** [comment]
@MarioCarrilloA can you check if this matches your thoughts, and also if it works under you CI setup.
Note, the density tests won't work until #410 lands - you could hand-copy over or comment out to test etc.
/cc @bwarden so you can see where this is heading, and cast your script eyes over it

/cc @mcastelino @amshinde @egernst - I cherry picked out some of the network tests to use for our baseline CI - pretty much looking at latency, bandwidth, jitter for TCP and some UDP. We can tweak this as we go along of course, but I thought I'd give you a pre heads-up.

Note to all - we do not want to run all the metrics for the CI - we want to focus on things that ideally don't take too long (but storage and some network tests will defeat that goal just because that is how they have go be) but do cover our basic requirements.

**Comment by MarioCarrilloA** [comment]
Hi @grahamwhaley, yes let's take a look :)

**Comment by grahamwhaley** [comment]
Oh, I should note @MarioCarrilloA - some of these tests (particularly the network ones iirc) will generate multiple results (such as read and write bandwidth or bandwidth and jitter) mixed into a single csv file. Today that will not work with `checkmetrics`, but we have two ways to fix that:
- short term, I think we put some `grep xxx.csv > yyy.csv` into this script as a temporary workaround - so we 'cherry pick' out the results into separate file so the `checkmetrics` toml file can be pointed at them
- longer term, I'd like to add the ability to `checkmetrics` so that it can filter the results out of the .csv files by itself. In my mind I have a [filter=[regexp]] sort of toml entry in the `checkmetrics` config files in mind.

File: .ci/run_metrics_ci.sh [hunk 1]
```diff
@@ -33,9 +35,59 @@ done
 pushd "$CURRENTDIR/../metrics"
 	source "lib/common.bash"
 
+	# If KSM is available on this platform, let's run the KSM tests first
+	# and then turn it off for the rest of the tests, as KSM may introduce
+	# some extra noise in the results by stealing CPU time for instance
+	if [ -f ${KSM_ENABLE_FILE} ]; then
```
Comment (bwarden) [inline .ci/run_metrics_ci.sh]: You can use `[[` here instead of calling the `[` program

Comment (grahamwhaley) [inline .ci/run_metrics_ci.sh]: heh - fixed

File: .ci/run_metrics_ci.sh [hunk 2]
```diff
@@ -33,9 +35,59 @@ done
...
+	bash network/network-metrics-iperf3.sh
+
+	# UDP bandwidths and packet loss
+	bash network/network-metrics-nuttcp.sh
+
+
+	#
+	# Run some IO tests
+	#
+	storage/fio_job.sh -b 16k -o randread -t "storage IO random read bs 16k"
```
Comment (grahamwhaley) [inline .ci/run_metrics_ci.sh]: ah, indeed! fixed

**Comment by grahamwhaley** [comment]
new version pushed. DNM label dropped (although, Mario, if you can confirm the test runs in your infra that would be nice before a merge, but not critical :-) )

**Comment by MarioCarrilloA** [comment]
lgtm

**Comment by chavafg** [comment]
lgtm

**Pull request merged**